packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts

```diff
@@ -357,11 +357,22 @@ export class Model extends ModelBase {
   }
 
   private renderContainer(container: ContainerDefinition): CfnModel.ContainerDefinitionProperty {
+    const image = container.image.bind(this, this);
+    const modelDataConfig = container.modelData?.bind(this, this);
+    const useModelDataSource = modelDataConfig?.compressionType === CompressionType.NONE
+     || modelDataConfig?.s3DataType === S3DataType.S3_PREFIX;
     return {
-      image: container.image.bind(this, this).imageName,
+      image: image.imageName,
       containerHostname: container.containerHostname,
       environment: container.environment,
-      modelDataUrl: container.modelData ? container.modelData.bind(this, this).uri : undefined,
+      modelDataSource: useModelDataSource ? {
+        s3DataSource: {
+          s3Uri: modelDataConfig.uri,
+          s3DataType: modelDataConfig.s3DataType!,
+          compressionType: modelDataConfig.compressionType!,
+        },
+      } : undefined,
+      modelDataUrl: !useModelDataSource ? modelDataConfig?.uri : undefined,
```

Comment:
Would it be possible for both of these values to be set at the same time? And if so, what sort of behaviour would we expect?